src/plugins/data/public/query/saved_query/saved_query_service.ts

```diff
@@ -143,20 +144,22 @@ export const createSavedQueryService = (
     id: string;
     attributes: SerializedSavedQueryAttributes;
   }) => {
-    let queryString;
+    let parsedQuery;
     try {
-      queryString = JSON.parse(savedQuery.attributes.query.query);
+      parsedQuery = JSON.parse(savedQuery.attributes.query.query);
     } catch (error) {
-      queryString = savedQuery.attributes.query.query;
+      parsedQuery = savedQuery.attributes.query.query;
     }
+
     const savedQueryItems: SavedQueryAttributes = {
       title: savedQuery.attributes.title || '',
       description: savedQuery.attributes.description || '',
       query: {
-        query: queryString,
+        query: isObject(parsedQuery) ? parsedQuery : savedQuery.attributes.query.query,
```

Comment:
There's some redundancy here that I think can be cleaned up.  

`savedQuery.attributes.query.query` is used three times. Can we define `queryString` as

`const queryString = savedQuery.attributes.query.query` then we can make use of the parsedQuery 
as we try to
```
try {
  parsedQuery = JSON.parse(queryString)
  parsedQuery = isObject(parsedQuery) ? parsedQuery : queryString
 catch (error) {
   parsedQuery = queryString;
 }
```

then `query` can just be updated to take parsedQuery

Reply:
I see. will fix to avoid redundancy.

Reply:
The original issue mentioned two cases which would cause issues
- `'\"Men's Shoes\"`
- `'\"Men's Shoes\" OR \"Men's Clothing\"`

both the existing code and new code both handle the second case identically. This is also the usecase in which the parse error is thrown. Is that intentional?